TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.CfgSentence.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using TypeCobol.Compiler.Symbols;
+
+namespace TypeCobol.Analysis.Cfg
+{
+    public partial class ControlFlowGraphBuilder<D>
+    {
+        /// <summary>
+        /// A Sentence used for our builder. A Sentence is a special symbol.
+        /// </summary>
+        internal class CfgSentence : Symbol
+        {
+            /// <summary>
+            /// First Block associated to this sentence.
+            /// </summary>
+            internal BasicBlockForNode Block
+            {
+                get;
+                set;
+            }
+
+            /// <summary>
+            /// The Block Index Associated to the Block.
+            /// </summary>
+            internal int BlockIndex
```

Comment:
=> Block.Index ?

Reply:
Ok, this the index in the Cfg.SuccessorEdges list of the Sentence's block. In other words the previous block before the sentence's block will have in its successor list a value which is the BlockIndex. So I agree the name of this field should be something like 'BlockIndexInSuccessorList'.
This value is also used for resolving target successor index for GOTOs

Reply:
See PR #1738. You're right, it's not the index of the block but its index in the successors list.